apps/trading/components/asset-option/asset-option.tsx

```diff
@@ -24,9 +28,68 @@ export const AssetOption = ({
             : asset.source.__typename}
         </div>
       </div>
-      <div className="ml-auto text-sm">
-        {addDecimalsFormatNumber(asset.balance, asset.decimals)}
-      </div>
+      {balance && (
+        <div className="ml-auto text-sm">
+          {addDecimalsFormatNumber(balance, asset.decimals)}
+        </div>
+      )}
     </div>
   );
 };
+
+type AssetBalanceProps = { chainId: string; address: string };
+
+/** Show the balance of an asset on an evm chain */
+export const AssetBalance = (props: AssetBalanceProps) => {
+  if (isAssetNative(props.address)) {
+    return <NativeTokenBalance {...props} />;
+  }
+
+  return <TokenBalance {...props} />;
+};
+
+const NativeTokenBalance = (props: AssetBalanceProps) => {
+  const { address } = useAccount();
+
+  const { data } = useBalance({
+    address,
+    chainId: Number(props.chainId),
+  });
+
+  if (!data) return null;
+
+  return <>{addDecimalsFormatNumber(data.value.toString(), data.decimals)}</>;
+};
+
+const TokenBalance = (props: AssetBalanceProps) => {
+  const { address } = useAccount();
+
+  const tokenAddress = props.address as `0x${string}`;
```

Comment:
suggestion: `0x${string}` is used in couple of places how about making it a common type e.g. `HexString`